config/shared_config_test.go

```diff
@@ -218,6 +218,33 @@ func TestNewSharedConfig(t *testing.T) {
 				},
 			},
 		},
+		"AWS SSO Profile": {
+			Filenames: []string{testConfigFilename},
+			Profile:   "sso_creds",
+			Expected: SharedConfig{
+				Profile:      "sso_creds",
+				SSOAccountID: "012345678901",
+				SSORegion:    "us-west-2",
+				SSORoleName:  "TestRole",
+				SSOStartURL:  "https://127.0.0.1/start",
+			},
+		},
+		"Assume Role with AWS SSO Credentials": {
+			Filenames: []string{testConfigFilename},
+			Profile:   "source_sso_creds",
+			Expected: SharedConfig{
+				Profile:           "source_sso_creds",
+				RoleARN:           "source_sso_creds_arn",
+				SourceProfileName: "sso_creds",
+				Source: &SharedConfig{
+					Profile:      "sso_creds",
+					SSOAccountID: "012345678901",
+					SSORegion:    "us-west-2",
+					SSORoleName:  "TestRole",
+					SSOStartURL:  "https://127.0.0.1/start",
+				},
+			},
+		},
```

Comment:
Might be redundant, but helpful to add test cases for profiles with:

1.  Chained Assume role with AWS SSO Credentials. 
2. Profile with SSO creds modeled together with static credentials.
3. Partial SSO creds is covered in a test below, so may or may not be added here.

Reply:
1. Is covered by the `source_sso_creds` profile test which uses a profile with a role_arn and a source profile that is SSO credentials.
2. Good call out, since this should favor the SSO credentials,
3. Can add this to just to ensure that the error surfaces correctly.

Reply:
1. I was thinking of multi-level chaining with assume role. Since credentials take precedence starting the second level. 

Basically looking for what's the behavior if we go through the following flow : 

```assume role -> assume role + sso creds -> assume role + creds ```

Reply:
I was unclear with `assume role` expectation, I should have written `assume role with source profile`.